### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by jerryneedell**
This is the initial version of a driver for the STMPE 610 Resistive Touch Screen Controller.

It supports both SPI and I2C.

For SPI - I have not yet addressed the issue reported by @ladyada  where on some occasions, the wrong clock edge gets used. I have only seen this very rarely and I'm still thinking about how to handle it. Usually, just retrying fixes it.

as suggested, this is patterned "loosely" after the FocalTouch driver.

For the "touches" method, I arbitrarily allow it to return up to 4 touches at at time. The controller buffer can hold 128, but it is easy to run out of memory. I the paint demo, it never needs to buffer more than 1 anyway.  Not sure if the buffering is useful or not, put it in as an example.

I have tested it on a 2.4inch TFT featherwing (SPI) and on a feather_m0_express with an stmpe610 breakout and touchscreen (PID 333) with SPI and I2C.

The paint demo has been tested on a 2.4 inch TFT Featherwing via a feather52 but the code in the repo is set up for feather_m0_express pins.

Comments welcome.

### CODE REVIEW

File: adafruit_stmpe610.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        # Subclasses MUST implement this!
+        raise NotImplementedError
+
+    def _write_register_byte(self, register, value):
+        # Write a single byte register at the specified register address.
+        # Subclasses MUST implement this!
+        raise NotImplementedError
+
+
+    # pylint: disable=unused-variable
```
Comment (tannewt) [inline adafruit_stmpe610.py]: What is unused?

File: adafruit_stmpe610.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+STMPE_TSC_CFG_DELAY_10MS = const(0x30)
+STMPE_TSC_CFG_DELAY_50MS = const(0x38)
+STMPE_TSC_CFG_SETTLE_10US = const(0x00)
+STMPE_TSC_CFG_SETTLE_100US = const(0x01)
+STMPE_TSC_CFG_SETTLE_500US = const(0x02)
+STMPE_TSC_CFG_SETTLE_1MS = const(0x03)
+STMPE_TSC_CFG_SETTLE_5MS = const(0x04)
+STMPE_TSC_CFG_SETTLE_10MS = const(0x05)
+STMPE_TSC_CFG_SETTLE_50MS = const(0x06)
+STMPE_TSC_CFG_SETTLE_100MS = const(0x07)
```
Comment (tannewt) [inline adafruit_stmpe610.py]: Do all of these need to be public? If you prefix them with _ they'll be omitted from the mpy and save memory when loaded.

File: examples/stmpe610_simpletest.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import busio
+import board
+import digitalio
+from adafruit_stmpe610 import Adafruit_STMPE610_SPI
+spi=busio.SPI(board.SCK,board.MOSI,board.MISO)
+cs = digitalio.DigitalInOut(board.D6)
+st=Adafruit_STMPE610_SPI(spi,cs)
+print("Go Ahead - Touch the Screen - Make My Day!")
```
Comment (tannewt) [inline examples/stmpe610_simpletest.py:8]: :-)

File: adafruit_stmpe610.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+"""
+
+# imports
+
+import time
+from micropython import const
+
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_stmpe610.git"
```
Comment (tannewt) [inline adafruit_stmpe610.py]: I believe STMPE should be capitalized to match the github repo name.

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+* `Register <https://github.com/adafruit/Adafruit_CircuitPython_Register>`_
+
+Please ensure all dependencies are available on the CircuitPython filesystem.
+This is easily achieved by downloading
+`the Adafruit library and driver bundle <https://github.com/adafruit/Adafruit_CircuitPython_Bundle>`_.
+
+Usage Example
+=============
+
+see examples in githup repository: https://github.com/adafruit/Adafruit_CIrcuitPython_STMPE610/exmaples
```
Comment (tannewt) [inline README.rst]: Mis-capitalization in `CIrcuitPython`

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks @jerryneedell ! I added a few comments.

File: adafruit_stmpe610.py (hunk 4 above)
Comment (jerryneedell) [reply adafruit_stmpe610.py]: Thanks - will change

File: README.rst (hunk 5 above)
Comment (jerryneedell) [reply README.rst]: argh. Thanks.

**Review by tannewt — APPROVED** [review]

**Comment by tannewt** [comment]
Thank you @jerryneedell !

**Pull request merged**